docs/source/transaction_family_specifications/validator_registry_transaction_family.rst

```diff
@@ -0,0 +1,257 @@
+protocol buffers code:
+
+.. code-block:: protobuf
+
+  message ValidatorRegistryPayload {
+    // The action that the transaction processor will take. Currently this
+    // is only “register”, but could include other actions in the futures
+    // such as “revoke”
+  	string verb = 1;
+
+    // The name of the endpoint. It should be a human readable name.
+  	string name = 2;
+
+
+  	// validator's public key (currently using signer_pubkey as this is
+    // stored in the transaction header)
+      string id = 3;
+
+    // This is the protocol buffer described above.
+    SignUpInfo signup_info = 4;
+
+    // The current chain head’s block num. This is used as a form of time
+    // to help with the enforcement of the C and K tests
+    int32 block_num = 5:
+
+  }
+
+
+
+Transaction Header
+==================
+
+Inputs and Outputs
+------------------
+
+The inputs for validator registry family transactions must include:
+
+* the address of *validator_id*
+* the address of *validator_map*
+
+The outputs for validator registry family transactions must include:
+
+* the address of *validator_id*
+* the address of *validator_map*
+
+
+Dependencies
+------------
+
+None
+
+Family
+------
+- family_name: "sawtooth_validator"
+- family_version: "1.0"
+
+
+Encoding
+--------
+
+The encoding field must be set to "application/protobuf".
+
+
+Execution
+=========
+
+Untrusted python code, that is a part of the transaction processor, will verify
+the attestation verification report for the signup information. It is important to
+note, that the IAS keys will need to be on the block chain and it will need to
+be set on configuration. This will allow both the simulator logic and real
+SGX logic to be the same.
+
+Before the validator_registry transaction can be sent the SignUpInfo needs to be
+created by the poet_consensus and it  will need to request the current
+chain_head’s block number from the journal.
+
+The journal will need a new method to retrieve this information.
+
+If the validator_name does not match syntactic requirements, the transaction is
+invalid. The current requirement is that the validator_name is 64 characters
+or less.
+
+If the validator_id does not match the transaction signer, the transaction is
+invalid. The validator_id should be the same as the signer_pubkey stored in the
+transaction header.
+
+The signup info needs to be verified next. The signup info, public key hash, and
+the most recent wait_certificate_id are passed to the enclave where the
+following checks are performed:
+
+- Verify the signature of the attestation verification report that is
+  stored in the signup_info.proof_data.
+- The verification report’s has a PSE manifest status and it must be OK.
+- PSE manifest hash exists and is value we expect.
+- The enclave quote is OK and contains a report body with the value we expect.
+- The wait_certificate_id stored in the nonce field matches the provided
+  wait_certificate_id.
```

Comment:
Not certain if you need to provide this level of detail for the validator registry as I am not certain how static these tests are....